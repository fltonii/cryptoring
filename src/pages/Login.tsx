import * as React from "react";
import { Box, TextInput, Text, Button } from "grommet";
import { ContentWrapper } from "../components/";
import { useSetTitle } from "../hooks";
import { withRouter } from "react-router-dom";

const Login = withRouter(({ history }) => {
  const [formVal, setFormVal] = React.useState({
    username: "",
    password: ""
  });

  useSetTitle("Login");

  const handleChange = (e: React.FormEvent<HTMLInputElement>) => {
    console.log(formVal);
    setFormVal({ ...formVal, [e.currentTarget.name]: e.currentTarget.value });
  };

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    console.log(e);
    history.push("/");
  };

  return (
    <ContentWrapper>
      <Box
        direction="column"
        align="center"
        justify="center"
        background="light-2"
        pad={{ vertical: "small", horizontal: "medium" }}
        elevation="medium"
        width="30vw"
        height="50vh"
      >
        <Text>Login</Text>
        <form onSubmit={(e: React.FormEvent<HTMLFormElement>) => onSubmit(e)}>
          <TextInput
            name="username"
            placeholder="username"
            value={formVal.username}
            onChange={(e: React.FormEvent<HTMLInputElement>) => handleChange(e)}
          />
          <br />
          <TextInput name="password" placeholder="password" />
          <br />
            <Button type="submit" color="main" hoverIndicator label="Log In!" />
        </form>
        <pre>{JSON.stringify(formVal)}</pre>
      </Box>
    </ContentWrapper>
  );
});

export default Login;
