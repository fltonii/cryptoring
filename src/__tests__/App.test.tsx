import * as React from "react";
import { render, fireEvent } from "react-testing-library";
import Login from "../pages/Login";

jest.mock("react-router-dom", () => ({
  withRouter: jest.fn().mockImplementation(component => component)
}));

test("Login snapshot", () => {
  const { container } = render(<Login />);
  expect(container.firstChild).toMatchSnapshot();
});

test("onSubmit gets called by button", () => {
  const { getByPlaceholderText, getByText } = render(<Login />);
  const usrname = getByPlaceholderText(/username/i) as HTMLInputElement;
  const pwrd = getByPlaceholderText(/password/i) as HTMLInputElement;
  const submit = getByText(/log in/i);
  usrname.value = "fltonii";
  pwrd.value = "123456";

  expect(usrname.value).toEqual("fltonii");
  expect(pwrd.value).toEqual("123456");

  submit.click();
});
