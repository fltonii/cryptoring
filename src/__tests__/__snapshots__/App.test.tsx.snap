// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<div
  class="sc-bdVaJa buonmG"
>
  <div
    class="StyledBox-sc-13pk1d4-0 cxDOqF"
  >
    <span
      class="StyledText-sc-1sadyjn-0 fWSbXS"
    >
      Login
    </span>
    <form>
      <div
        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
      >
        <input
          autocomplete="off"
          class="StyledTextInput-sc-1x30a0s-0 iQGivQ"
          name="username"
          placeholder="username"
          value=""
        />
      </div>
      <br />
      <div
        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
      >
        <input
          autocomplete="off"
          class="StyledTextInput-sc-1x30a0s-0 iQGivQ"
          name="password"
          placeholder="password"
          value=""
        />
      </div>
      <br />
      <button
        class="StyledButton-sc-323bzc-0 jcunWI"
        type="submit"
      >
        Log In!
      </button>
    </form>
    <pre>
      {"username":"","password":""}
    </pre>
  </div>
</div>
`;
